server.go

```diff
@@ -1944,6 +1961,40 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets a compressor for outbound messages.
+// It must not be called after any event that causes headers to be sent
+// (see SetHeader for a complete list). Provided compressor is used when below
```

Comment:
Should we check to see if headers have already been sent and return an error if so?

Reply:
We do this already in the `stream.SetSendCompress` method, which checks if the header is sent or stream is done